cyfra-core/src/main/scala/io/computenode/cyfra/core/GProgram.scala

```diff
@@ -0,0 +1,47 @@
+package io.computenode.cyfra.core
+
+import io.computenode.cyfra.core.layout.LayoutStruct
+import io.computenode.cyfra.dsl.gio.GIO
+import io.computenode.cyfra.core.layout.Layout
+
+import java.nio.ByteBuffer
+import GProgram.*
+import io.computenode.cyfra.dsl.Value
+import io.computenode.cyfra.dsl.buffer.GBuffer
+import io.computenode.cyfra.dsl.struct.GStruct
+import io.computenode.cyfra.dsl.struct.GStruct.Empty
+import izumi.reflect.Tag
+
+class GProgram[Params, Uniform <: GStruct[Uniform], L <: Layout: LayoutStruct] private (
+  val body: (L, Uniform) => GIO[?],
+  val layout: InitProgramLayout => Params => L,
+  val uniform: Params => Uniform,
```

Comment:
1. Only one uniform per program?
2. Can `GStruct` be of unknown size?
3. Can `Uniform` be part of a `Layout`

Reply:
GStruct can be of various size. One Uniform as a struct simplifies it a bit, if it's enough to express all we need it may make sense. This or it would be just a special kind of GBuffer in Layout